assets/js/components/ProjectEditor.js

```diff
@@ -0,0 +1,245 @@
+    }
+
+    window.history.pushState(
+      { type: 'ProjectEditor', id: programId, full: true },
+      $(this).text(),
+      '#' + programId
+    )
+
+    $(window).on('popstate', this.popStateHandler)
+    showCustomTopBarTitle('', function () {
+      window.history.back()
+    })
+
+    this.body.addClass('overflow-hidden')
+    this.editTextUI.removeClass('d-none')
+  }
+
+  this.close = () => {
+    $(window).off('popstate', this.popStateHandler)
+    showDefaultTopBarTitle()
+
+    if (this.reloadScreen) {
+      window.location.reload()
+    }
+
+    this.body.removeClass('overflow-hidden')
+    this.editTextUI.addClass('d-none')
+    this.reset()
+  }
+
+  this.reset = () => {
+    this.languageSelect.selectedIndex = 0
+    this.previousIndex = 0
+    this.getNewText()
+  }
+
+  function getLanguages () {
+    $.ajax({
+      url: '../languages',
+      type: 'get',
+      success: function (data) {
+        self.languages = data
+        self.populateSelector()
+      }
+    })
+  }
+
+  this.populateSelector = () => {
+    this.languageSelectorList.empty()
+    this.languageSelectorList.append('<li class="mdc-list-item" data-value="default" role="option" tabindex="-1">' +
+      '<span class="mdc-list-item__ripple"></span>' +
+      '<span class="mdc-list-item__text">' + this.defaultText + '</span>' +
+      '</li>')
+
+    for (const language in this.languages) {
+      if (language.length <= 2) {
+        this.languageSelectorList.append(`<li class="mdc-list-item" data-value="${language}" role="option" tabindex="-1">\
+          <span class="mdc-list-item__ripple"></span>\
+          <span class="mdc-list-item__text">${this.languages[language]}</span>\
+          </li>`)
+      }
+    }
+
+    this.languageSelect.layoutOptions()
+    this.reset()
+  }
+
+  this.areChangesSaved = () => {
+    for (const textField of this.textFields) {
+      if (!textField.areChangesSaved()) {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  function keepOrDiscardChangesResult (result) {
+    if (result.isConfirmed) {
+      self.previousIndex = self.languageSelect.selectedIndex
+      for (const textField of self.textFields) {
+        textField.keepChanges(self.languageSelect.value)
+      }
+    } else if (result.isDenied) {
+      self.previousIndex = self.languageSelect.selectedIndex
+      self.getNewText()
+    } else if (result.isDismissed) {
+      self.languageSelect.selectedIndex = self.previousIndex
+    }
+  }
+
+  function closeEditorDialogResult (result) {
+    if (result.isConfirmed) {
+      self.save()
+    } else if (result.isDenied) {
+      self.close()
+    }
+  }
+
+  this.getSectionText = (sections) => {
+    let sectionText = ''
+    for (let i = 0; i < sections.length; i++) {
+      if (i === 0) {
+        sectionText += sections[i]
+      } else if (i < sections.length - 1) {
+        sectionText += ', ' + sections[i] + ','
+      } else {
+        sectionText += ' and ' + sections[i]
```

Comment:
Need to make translatable

Reply:
hold on. it doesn't have to include the sections. it could just say "changes saved" or like "changes for <language> saved"

Reply:
Okay, so in the case where a user deletes the translation for the description and updates the translation for the credits, there should be a notification saying translation saved and another one saying translation deleted? Or just one notification saying translation updated?

Reply:
I think one snackbar is enough, since everything is successful, there's no more action required from the user. plus these two snackbars are going to be shown on top of each other?

Reply:
Okay, combined to one snackbar that says that the translation is updated.

Also, from my testing, it looks like the snackbars show one after the other. Might be a timing thing though so they could potentially show at the same time.